core/src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -178,10 +178,10 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     public QueryValidationException validate() {
         QueryValidationException validationException = null;
         if (fieldName == null || fieldName.isEmpty()) {
-            validationException = QueryValidationException.addValidationError("field name cannot be null or empty.", validationException);
+            validationException = QueryValidationException.addValidationError(getName(), "field name cannot be null or empty.", validationException);
```

Comment:
good point, this makes sense. I just checked what we currently do given that validation before refactoring happens while parsing. No context is given for messages that we can move to validate, which are not json specific. That means this is not a regression caused by the refactoring, rather an improvement that we could work on in the future ;)